rust/src/operations/write.rs

```diff
@@ -0,0 +1,458 @@
+            } => {
+                let uri = table_uri.into();
+                let plan = Arc::new(WritePartitionCommand::new(
+                    uri.clone(),
+                    partition_by.clone(),
+                    input.clone(),
+                ));
+                Ok(Self {
+                    table_uri: uri,
+                    mode,
+                    partition_columns: partition_by,
+                    predicate,
+                    schema: input.schema(),
+                    input: plan,
+                })
+            }
+            _ => Err(DeltaCommandError::UnsupportedCommand(
+                "WriteCommand only implemented for write operation".to_string(),
+            )),
+        }
+    }
+}
+
+#[async_trait]
+impl ExecutionPlan for WriteCommand {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> ArrowSchemaRef {
+        Arc::new(ArrowSchema::new(vec![ArrowField::new(
+            "serialized",
+            DataType::Utf8,
+            true,
+        )]))
+    }
+
+    fn children(&self) -> Vec<Arc<dyn ExecutionPlan>> {
+        vec![self.input.clone()]
+    }
+
+    fn output_partitioning(&self) -> Partitioning {
+        Partitioning::UnknownPartitioning(1)
+    }
+
+    fn output_ordering(&self) -> Option<&[PhysicalSortExpr]> {
+        None
+    }
+
+    fn required_child_distribution(&self) -> Distribution {
+        Distribution::UnspecifiedDistribution
+    }
+
+    fn with_new_children(
+        &self,
+        _children: Vec<Arc<dyn ExecutionPlan>>,
+    ) -> DataFusionResult<Arc<dyn ExecutionPlan>> {
+        todo!()
+    }
+
+    async fn execute(
+        &self,
+        partition: usize,
+        context: Arc<TaskContext>,
+    ) -> DataFusionResult<SendableRecordBatchStream> {
+        let mut table =
+            get_table_from_uri_without_update(self.table_uri.clone()).map_err(to_datafusion_err)?;
+        let metrics = ExecutionPlanMetricsSet::new();
+        let tracking_metrics = MemTrackingMetrics::new(&metrics, partition);
+
+        let mut actions = match table.load().await {
+            // Table is not yet created
+            Err(_) => {
+                let schema = Schema::try_from(self.schema.clone()).map_err(to_datafusion_err)?;
+                let metadata = DeltaTableMetaData::new(
+                    None,
+                    None,
+                    None,
+                    schema,
+                    self.partition_columns.clone().unwrap_or_default(),
+                    HashMap::new(),
+                );
+                let op = DeltaOperation::Create {
+                    location: self.table_uri.clone(),
+                    metadata: metadata.clone(),
+                    mode: SaveMode::ErrorIfExists,
+                    // TODO get the protocol from somewhere central
+                    protocol: Protocol {
+                        min_reader_version: 1,
+                        min_writer_version: 2,
```

Comment:
Technically, since you aren't enforcing `appendOnly` or Column invariants, this should be 1 for now.

In addition, I think you should be checking the protocol version of an existing table and erroring if it's > 1.

Reply:
absolutely. updated min version and added chack for exisiting table.